mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxPath.java

```diff
@@ -48,6 +52,10 @@ public static MailboxPath forUser(String username, String mailboxName) {
         return new MailboxPath(MailboxConstants.USER_NAMESPACE, username, mailboxName);
     }
 
+    private static final String INVALID_CHARS = "%*&#";
```

Comment:
From the RFC i have seen only that the use of `%` and `*` are discouraged.
What is the reason we forbid to use '%' and `#` ?

Reply:
The RFC states:

```
   5)    Two characters, "#" and "&", have meanings by convention, and
         should be avoided except when used in that convention.
```

`#` is a special character identifying fully qualified names as oposed to relatives names. This is the namespace prefix.

Regarding `&` I would bet this is related to modified UTF-7 - it's use would require potentially buggy escaping strategies.

```
   "&" is used to shift to modified BASE64 and "-" to shift back to
   US-ASCII.  There is no implicit shift from BASE64 to US-ASCII, and
   null shifts ("-&" while in BASE64; note that "&-" while in US-ASCII
   means "&") are not permitted.  However, all names start in US-ASCII,
   and MUST end in US-ASCII; that is, a name that ends with a non-ASCII
   ISO-10646 character MUST end with a "-").
```